include/path_tracking_pid/details/derivative.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <path_tracking_pid/details/fifo_array.hpp>
+
+namespace path_tracking_pid::details
+{
+/**
+ * @brief discrete time derivative filter
+ */
+class Derivative
+{
+public:
+  /**
+   * @brief Construct a Derivative instance
+   */
+  Derivative();
+
+  /**
+   * @brief filter one sample of a signal
+   * @param u signal to be filtered
+   * @param step_size
+   * @return derivative of the signal
+   */
+  double filter(double u, double step_size);
+
+  void reset();
```

Comment:
This does not do anything? Is that intentional? If so, why not just remove it?

If not, consider documenting it.

Reply:
It's a mistake, it should clear the previous sample

Reply:
Fixed